### PR TITLE
FS-4008: Full application comments

### PR DESCRIPTION
**Pull request by hamzabinkhalid**
### Ticket
https://dluhcdigital.atlassian.net/browse/FS-4008

### Change description
- Added a new enum WHOLE_APPLICATION to handle the logic of application comments section
- Migrated and upgraded the DB
- Updated the method names to reflect the changes closely

### CODE REVIEW

File: db/migrations/versions/~2024_04_15_1844-5d3468e62492_add_whole_application_to_commenttype.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+enum_name = "commenttype"
+# Define the old options
+old_options = ("COMMENT",)
+# Define the new options
+new_options = ("COMMENT", "WHOLE_APPLICATION")
+# Create the old enum type
+old_type = sa.Enum(*old_options, name=enum_name)
+# Create the new enum type
+new_type = sa.Enum(*new_options, name=enum_name)
+
```
Comment (srh-sloan) [inline db/migrations/versions/~2024_04_15_1844-5d3468e62492_add_whole_application_to_commenttype.py:28]: Nice to have these defined up here for reuse in `upgrade()` and `downgrade()` 👍

File: tests/test_db_function.py [hunk 2]
```diff
@@ -184,27 +184,27 @@ def test_get_comments(seed_application_records):
         "user_id": "test",
         "theme_id": theme_id,
     }
-    create_comment_for_application_sub_crit(**assessment_payload_2)
+    create_comment(**assessment_payload_2)
 
     assessment_payload_3 = {
         "application_id": application_id,
         "sub_criteria_id": sub_criteria_id,
```
Comment (srh-sloan) [inline tests/test_db_function.py:191]: Can we make this `None` to match how the code uses `WHOLE_APPLICATION` pls?

**Review by srh-sloan — CHANGES_REQUESTED** [review]
Looking good, but made a couple of comments. Also, we haven't renamed any of the `put_comment...` methods or tested those with our new comment type - do we use those `put` methods to update whole application comments?

File: tests/test_db_function.py (hunk 2 above)
Comment (srh-sloan) [reply tests/test_db_function.py:191]: And `theme_id`

Comment (srh-sloan) [reply tests/test_db_function.py:191]: Also, can we put back what was there previously and create a new test case for the this new scenario pls

Comment (hamzabinkhalid) [reply tests/test_db_function.py]: done

**Review by tferns — COMMENTED** [review]
Nice work, can you make sure pre-commit ci is passing though?

**Comment by hamzabinkhalid** [comment]
> > Nice work, can you make sure pre-commit ci is passing though?
> 
> @hamzabinkhalid ^

done

**Review by tferns — APPROVED** [review]

**Review by srh-sloan — APPROVED** [review]
Looking good, nice job 👍

**Pull request merged**